include/ts/ts.h

```diff
@@ -1778,6 +1778,14 @@ tsapi void TSHttpTxnServerIntercept(TSCont contp, TSHttpTxn txnp);
     @param addr Target address of the origin server.
     @param tag A logging tag that can be accessed via the pitag field. May be @c NULL.
     @param id A logging id that can be access via the piid field.
+    @param buffer_index Index for read and write buffers used by PluginVC.
+    @param buffer_water_mark Number of bytes used as the water mark for buffers used by PluginVC.
+ */
+tsapi TSVConn TSHttpConnectPlugin(struct sockaddr const *addr, const char *tag, int64_t id, int64_t buffer_index,
+                                  int64_t buffer_water_mark);
```

Comment:
Why isn't this documented in the TS manual?

Reply:
I just pushed a commit to address this; please take a look at your convenience.

Reply:
Should that buffer_index value have enumeration/int64_t values defined in apidefs.h that parallel these values in core?  There are 15 total values:
BUFFER_SIZE_INDEX_128
...
BUFFER_SIZE_INDEX_32K
...

Reply:
@traeak I just pushed a commit that introduces a struct (`TSHttpConnectPluginOptions`) to replace the function arguments, and the `buffer_index` member now matches the type you suggest. That said, I am only using that type in the API and its dependencies, and will update PR #8089 to do the same if these changes are accepted.

The changes I made to the PluginVC interface that exist inside the core will continue to use the `#defined` values; this seems to be in the spirit of how these two synonymous things are used in practice (API vs `#define`ed). Do you agree?